python/cuml/preprocessing/model_selection.py

```diff
@@ -45,26 +46,33 @@ def _stratify_split(X, y, n_train, n_test, x_numba, y_numba, random_state):
     """
     x_cudf = False
     y_cudf = False
+
     if isinstance(X, cudf.DataFrame):
         x_cudf = True
-        X = X.values
     elif hasattr(X, "__cuda_array_interface__"):
         X = cp.asarray(X)
+        x_order = _strides_to_order(X.__cuda_array_interface__['strides'],
+                                    cp.dtype(X.dtype))
+
     if isinstance(y, cudf.Series):
         y_cudf = True
-        y = y.values
     elif hasattr(y, "__cuda_array_interface__"):
         y = cp.asarray(y)
+        y_order = _strides_to_order(y.__cuda_array_interface__['strides'],
+                                    cp.dtype(y.dtype))
     elif isinstance(y, cudf.DataFrame):
         y_cudf = True
         # ensuring it has just one column
         if y.shape[1] == 1:
-            y = y.values
+            pass
         else:
             raise ValueError('Expected one label, but found y'
                              'with shape = %d' % (y.shape))
 
-    classes, y_indices = cp.unique(y, return_inverse=True)
+    if y_cudf:
+        classes, y_indices = cp.unique(y.values, return_inverse=True)
```

Comment:
Trivial point, but I would probably do:
```
 classes, y_indices = cp.unique(y.values if y_cudf else y, return_inverse=True)
```

Since that emphasizes that both codepaths are the same but with one small exception.

Reply:
Thanks, it looks much cleaner too!